src/libs/actions/IOU.ts

```diff
@@ -3072,6 +3072,7 @@ function trackExpense(
         }
     }
     if (action === CONST.IOU.ACTION.SHARE) {
+        Navigation.dismissModal();
```

Comment:
This will just close the modal without navigating the user to the report chat and will only navigate to the report chat after inviting the user (@jjcoffee as you mentioned [here](https://github.com/Expensify/App/issues/40490#issuecomment-2066067480))/go back from the invite page.

Do we want to navigate the user after submitting the expense? Here is how it looks:

https://github.com/Expensify/App/assets/50919443/7e522696-4099-4c4b-b834-0ee8f9dbf6d1

Reply:
Personally I think it makes sense to immediately navigate upon submitting the expense, but there may be some reason why this was not implemented and I'm not sure it's really within scope for this PR - it's more of a polish thing.

Reply:
Asked on Slack!